main.go

```diff
@@ -14,7 +15,386 @@
+	Payload               string            `default:"ETHERNET" desc:"Name of provided service payload" split_words:"true"`
+	Labels                map[string]string `default:"" desc:"Endpoint labels"`
+	DNSConfigs            dnstools.Decoder  `default:"[]" desc:"DNSConfigs represents array of DNSConfig in json format. See at model definition: https://github.com/networkservicemesh/api/blob/main/pkg/api/networkservice/connectioncontext.pb.go#L426-L435" split_words:"true"`
+	CidrPrefix            []string          `default:"169.254.0.0/16" desc:"List of CIDR Prefix to assign IPv4 and IPv6 addresses from" split_words:"true"`
+	IdleTimeout           time.Duration     `default:"0" desc:"timeout for automatic shutdown when there were no requests for specified time. Set 0 to disable auto-shutdown." split_words:"true"`
+	RegisterService       bool              `default:"true" desc:"if true then registers network service on startup" split_words:"true"`
+	PBRConfigPath         string            `default:"/etc/policy-based-routing/config.yaml" desc:"Path to policy based routing config file" split_words:"true"`
+	LogLevel              string            `default:"INFO" desc:"Log level" split_words:"true"`
+	OpenTelemetryEndpoint string            `default:"otel-collector.observability.svc.cluster.local:4317" desc:"OpenTelemetry Collector Endpoint"`
+}
+
+// Process prints and processes env to config
+func (c *Config) Process() error {
+	if err := envconfig.Usage("nsm", c); err != nil {
+		return errors.Wrap(err, "cannot show usage of envconfig nse")
+	}
+	if err := envconfig.Process("nsm", c); err != nil {
+		return errors.Wrap(err, "cannot process envconfig nse")
+	}
+	return nil
+}
+
 func main() {
+	// ********************************************************************************
+	// setup context to catch signals
+	// ********************************************************************************
+	ctx, cancel := signal.NotifyContext(
+		context.Background(),
+		os.Interrupt,
+		// More Linux signals here
+		syscall.SIGHUP,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+	defer cancel()
+
+	// ********************************************************************************
+	// setup logging
+	// ********************************************************************************
+	log.EnableTracing(true)
+	logrus.SetFormatter(&nested.Formatter{})
+	ctx = log.WithLog(ctx, logruslogger.New(ctx, map[string]interface{}{"cmd": os.Args[0]}))
+
+	if err := debug.Self(); err != nil {
+		log.FromContext(ctx).Infof("%s", err)
+	}
+
+	// enumerating phases
+	log.FromContext(ctx).Infof("there are 6 phases which will be executed followed by a success message:")
+	log.FromContext(ctx).Infof("the phases include:")
+	log.FromContext(ctx).Infof("1: get config from environment")
+	log.FromContext(ctx).Infof("2: retrieve spiffe svid")
+	log.FromContext(ctx).Infof("3: create icmp server ipam")
```

Comment:
We should fix the text here as its not icmp server.  I'd suggest just changing it to 'create ipam' (and below to 'create nse'